### ISSUE TITLE
test_update_grows_state only asserts shapes, hiding the underlying Stream.update y-data-drop bug

### ISSUE THREAD

**Issue by sachncs** [sachncs/laker#22]
## Problem

`test_update_grows_state` only asserts that `m.coef.shape[0] == n + 5` after `update`. It does not check the *values* of `coef`, so the underlying bug (where `update` loses the previous `y_train` and solves against a zero-padded RHS — see related issue on `Stream.update`) is invisible to CI. This is exactly the false-confidence pattern the harness warns about.

## Evidence

`tests/test_stream.py:10-23`:

```python
def test_update_grows_state(self):
    torch.manual_seed(0)
    n = 20
    x = torch.rand(n, 2, dtype=torch.float64) * 100
    y = torch.sin(x[:, 0] / 50)
    m = Laker(embed_dim=4, dtype=torch.float64, verbose=False)
    m.fit(x, y)
    assert m.coef.shape[0] == n

    x_new = torch.rand(5, 2, dtype=torch.float64) * 100
    y_new = torch.sin(x_new[:, 0] / 50)
    m.update(x_new, y_new, threshold=100, seed=0)
    assert m.coef.shape[0] == n + 5
    assert m.embed.shape[0] == n + 5
```

The two value-bearing assertions (`m.coef.shape[0]`, `m.embed.shape[0]`) are pure shape checks. The companion test `test_update_threshold_raises` does pass a `150`-row batch, but only asserts the exception is raised, not that the rest of the API still works.

The harness flags this as a "false confidence" pattern: line coverage is fine, but the test does not exercise the actual contract.

## Root Cause

The streaming test was added at a time when `update` was simpler (one-shot re-fit). After the public-API rename broke the underlying state lookup, the test continued to pass because shape is preserved regardless of the RHS content.

## Impact

The streaming workflow is silently broken in production, but CI stays green. New contributors will be misled into believing `update` works.

## Recommendation

Replace the shape-only assertions with value-level assertions. For example:

```python
m.update(x_new, y_new, threshold=100, seed=0)
assert m.coef.shape[0] == n + 5

# Value-level: the new model's predictions on the new points
# must be at least as accurate as the old model's predictions on them
y_new_pred_after = m.predict(x_new)
old_pred = m.predict(x_new)  # but we lost the old model — use coef[:n]
m_pred_old_on_new = (m.kernel.eval(m.embed[m.coef.shape[0] - len(y_new):],
                                   m.embed[:m.coef.shape[0] - len(y_new)],
                                   ) @ m.coef[:m.coef.shape[0] - len(y_new)])
assert torch.allclose(y_new_pred_after, m_pred_old_on_new, atol=...)
```

A simpler alternative: re-fit a fresh `Laker` on `torch.cat([x, x_new])` / `torch.cat([y, y_new])` and assert `torch.allclose` between the two `coef`s.

## Acceptance Criteria

- [ ] `test_update_grows_state` includes a value-level assertion (e.g. `coef[:n]` is close to the original `coef` after `update`).
- [ ] The streaming workflow is regression-tested with a deterministic seed.
- [ ] The new test fails against the current code and passes once the underlying `Stream.update` bug is fixed.

## Verification

Run `pytest tests/test_stream.py -x` and confirm the new assertion fails against the current code, then passes after the `Stream.update` fix.

## Dependencies

- Should be filed together with the `Stream.update` underscore-mismatch bug; the new test only passes once both are addressed.

## Risk

Low. Adding an assertion cannot regress a working API.

**Comment by sachncs** [comment]
Resolved in PR #32.
